packages/discovery-provider/plugins/pedalboard/apps/relay/src/middleware/validator.ts

```diff
@@ -50,34 +48,53 @@ export const validator = async (
 
   // Gather user from input data
   // @ts-ignore, partially populate for now
-  let user: Users = {
+  let recoveredSigner: Users | DeveloperApps = {
     wallet: senderAddress || null,
     handle: handle || null
   }
+
+  let signerIsApp = false
+  let signerIsUser = false
+
   try {
-    user = await retrieveUser(
+    recoveredSigner = await retrieveUser(
       contractRegistryKey,
       contractAddress,
       encodedABI,
       senderAddress,
       handle
     )
+    signerIsUser = true
   } catch (e) {
     logger.error(
       { e },
       'could not gather user from db, continuing with senderAddress and handle'
     )
   }
 
+  // could not find user
+  if (!signerIsUser) {
+    const developerApp = await retrieveDeveloperApp({ encodedABI, contractAddress })
+    if (developerApp === undefined) {
+      logger.error({ encodedABI }, "neither user nor developer app could be found for address")
+      validationError(next, 'recoveredSigner not valid')
```

Comment:
not sure whether this should 500 or 400 here, this is the case where there's no app or user for this address so i'm leaning towards bad request?

Reply:
i buy the bad request reasoning